src/api/posts.ts

```
@@ -0,0 +1,25 @@
+import { Post } from '../types';
+
+export const BASE_URL = 'https://mate.academy/students-api/';
+
+export async function getPosts(userId: string): Promise<Post[]> {
+  const response = await fetch(`${BASE_URL}/posts?userId=${userId}`);
+
+  if (!response.ok) {
+    throw new Error(`${response.status} - ${response.statusText}`);
+  }
+
+  return response.json();
+}
+
+export async function getPostbyId(postId:number) {
+  const response = await fetch(`${BASE_URL}/posts/${postId}`);
+
+  return response.json();
+}
+
```

The getPostbyId and getAllPosts functions are not checking if the API response is successful. This can result in silent errors if the API returns an error or a status other than 200

---

It is good practice to specify the return type for functions, especially when using TypeScript. This helps ensure that the code is safer and more understandable. The getPostbyId and getAllPosts functions should return the expected type, which is Promise<Post> for getPostbyId and Promise<Post[]> for getAllPosts.